src/AppBundle/v2/Registration/SelfRegistration/Factory/CasRecFactory.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace AppBundle\v2\Registration\SelfRegistration\Factory;
+
+use AppBundle\Entity\CasRec;
+use AppBundle\Service\DateTimeProvider;
+use AppBundle\v2\Registration\DTO\LayDeputyshipDto;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+class CasRecFactory
+{
+    /** @var ValidatorInterface */
+    private $validator;
+
+    /** @var DateTimeProvider */
+    private $dateProvider;
+
+    /**
+     * @param ValidatorInterface $validator
+     * @param DateTimeProvider $dateProvider
+     */
+    public function __construct(ValidatorInterface $validator, DateTimeProvider $dateProvider)
+    {
+        $this->validator = $validator;
+        $this->dateProvider = $dateProvider;
+    }
+
+    /**
+     * @param LayDeputyshipDto $dto
+     * @return CasRec
+     */
+    public function createFromDto(LayDeputyshipDto $dto)
+    {
+        $entity = new CasRec($this->convertDtoToArray($dto));
+        $entity->setUpdatedAt($this->dateProvider->getDateTime());
```

Comment:
What's the benefit of using a provider over `new \DateTime()`? Abstraction?

Reply:
The benefit comes with unit tests, being able to mock the response when making assertions on the value of a date. If source code uses `$v->setFoo(new \DateTime())` and a tests asserts `$this->assertEquals(new \DateTime(), $v->getFoo())`, it can cause sporadic test failures as the value of the `seconds` can differ between the two objects.

Reply:
See usage here: https://github.com/ministryofjustice/opg-digi-deps-api/pull/566/files#diff-eebdd052d15cc0753febeefb8efc2c4c

Reply:
Oh that's really smart, nice.